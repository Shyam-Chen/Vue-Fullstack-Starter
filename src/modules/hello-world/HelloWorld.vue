<script setup>
import { reactive } from 'vue';
import { useStore } from 'vuex';

// import state from './'

const store = useStore();
console.log(store.state);

// store.registerModule('helloWorld', {
//   namespaced: true,
// });

const state = reactive({ count: 0 });
</script>

<template>
  <div>{{ state.count }}</div>
</template>
