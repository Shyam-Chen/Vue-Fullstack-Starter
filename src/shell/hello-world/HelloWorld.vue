<template>
  <v-container id="hello-world" class="shell-container" fluid>
    <v-layout>
      <div class="title">{{ message }}</div>
    </v-layout>
  </v-container>
</template>

<script>
// @flow

type Data = {
  message: string,
};

export default {
  metaInfo: {
    title: 'Hello World | Oh My Vue',
    meta: [
      { property: 'og:title', content: 'Hello World | Oh My Vue' },
      { property: 'og:description', content: 'A bare-bones Hello World example.' },
      { name: 'twitter:title', content: 'Hello World | Oh My Vue' },
      { name: 'twitter:description', content: 'A bare-bones Hello World example.' },
    ],
  },
  data(): Data {
    return {
      message: 'Hello, World!',
    };
  },
};
</script>

<style scoped>
</style>
