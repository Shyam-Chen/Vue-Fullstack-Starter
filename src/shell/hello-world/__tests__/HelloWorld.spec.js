import { shallowMount } from '@vue/test-utils';

import HelloWorld from '../HelloWorld';

describe('HelloWorld', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(HelloWorld);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
