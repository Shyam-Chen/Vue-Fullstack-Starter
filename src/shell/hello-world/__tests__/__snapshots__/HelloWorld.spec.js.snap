// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render an initial component 1`] = `
<v-container-stub id="hello-world" tag="div" fluid="true" class="shell-container">
  <v-layout-stub tag="div">
    <div class="title">Hello, World!</div>
  </v-layout-stub>
</v-container-stub>
`;
